django/contrib/admin/options.py

```diff
@@ -1404,9 +1404,12 @@ def get_changeform_initial_data(self, request):
                 initial[k] = initial[k].split(",")
         return initial
 
-    @csrf_protect_m
-    @transaction.atomic
     def changeform_view(self, request, object_id=None, form_url='', extra_context=None):
```

Comment:
You should keep the `csrf_protect_m` decorators on the `change_form` and `delete_view` instead as preserves the actual nesting ordering and avoids beginning a transaction in the rare case CSRF validation fails.